### PR TITLE
Make MessagePollResponse more flexible

### LINKED ISSUE #59: Support more response types for MessagePollResponse
We need a way to support multiple response types for MessagePollResponse. Right now we expect a `DomainTransferResponseData` object but it can be many things.

For example the contents of `resData` might be a `HostInfoResponseData`:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<epp xmlns="urn:ietf:params:xml:ns:epp-1.0">
    <response>
        <result code="1301">
            <msg>Command completed successfully; ack to dequeue</msg>
        </result>
        <msgQ count="4" id="2728300">
            <qDate>2022-01-02T11:30:45Z</qDate>
            <msg>Unused objects policy</msg>
        </msgQ>
        <resData>
            <host:infData xmlns:host="urn:ietf:params:xml:ns:host-1.0">
                <host:name>ONE.NS.IDS11111111.COM</host:name>
                <host:roid>33661487_HOST_CNE-VRSN</host:roid>
                <host:status s="ok"/>
                <host:addr ip="v4">1.1.1.1</host:addr>
                <host:clID>29000333</host:clID>
                <host:crID>insdom1</host:crID>
                <host:crDate>2021-12-01T22:40:48Z</host:crDate>
                <host:upID>insdom1</host:upID>
                <host:upDate>2021-12-01T22:40:48Z</host:upDate>
            </host:infData>
        </resData>
        <extension>
            <changePoll:changeData state="before" xmlns:changePoll="urn:ietf:params:xml:ns:changePoll-1.0">
                <changePoll:operation op="purge">delete</changePoll:operation>
                <changePoll:date>2022-01-02T11:30:45Z</changePoll:date>
                <changePoll:svTRID>1641123045000-1774862978</changePoll:svTRID>
                <changePoll:who>regy_batch</changePoll:who>
                <changePoll:reason>Unused objects policy</changePoll:reason>
            </changePoll:changeData>
        </extension>
        <trID>
            <clTRID>abc</clTRID>
            <svTRID>123</svTRID>
        </trID>
    </response>
</epp>
```

### PR DESCRIPTION
**Pull request by nrempel**
Closes #59 

This pull request improve MessagePollResponse to support more response types.

I'm sure there are more response types that should be expected but I have not found a list anywhere.

### CODE REVIEW

File: src/message/poll.rs [hunk 1]
```diff
@@ -29,41 +30,24 @@ impl Default for MessagePoll<'static> {
...
-    /// The date by which the transfer request should be acknowledged
-    #[serde(rename = "domain:acDate", alias = "acDate")]
-    pub ack_by: StringValue<'static>,
-    /// The domain expiry date
-    #[serde(rename = "domain:exDate", alias = "exDate")]
-    pub expiring_at: Option<StringValue<'static>>,
+pub enum MessageData {
+    /// Data under the &lt;domain:trnData&gt; tag
+    #[serde(rename = "domain:trnData")]
+    DomainTransferResponseData(DomainTransferResponseData),
```
Comment (djc) [inline src/message/poll.rs:37]: Nice deduplication!

Maybe ditch the `ResponseData` suffix of the variant name here?

**Review by djc — APPROVED** [review]
LGTM!

File: src/message/poll.rs (hunk 1 above)
Comment (nrempel) [reply src/message/poll.rs]: Great idea

**Comment by masalachai** [comment]
Thanks! Merging.

**Pull request merged**